stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt

```diff
@@ -6,31 +6,62 @@ import android.graphics.Paint
 import android.graphics.Path
 import android.graphics.PixelFormat
 import android.graphics.drawable.Drawable
-import androidx.annotation.ColorInt
 import androidx.core.content.ContextCompat
 import com.getstream.sdk.chat.adapter.MessageListItem
 import com.getstream.sdk.chat.utils.Utils.dpToPx
 import io.getstream.chat.android.ui.R
 import io.getstream.chat.android.ui.messages.adapter.viewholder.MessageDeletedViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.MessagePlainTextViewHolder
 
 internal class BackgroundDecorator : BaseDecorator() {
 
     override fun decorateDeletedMessage(viewHolder: MessageDeletedViewHolder, data: MessageListItem.MessageItem) {
-        val context = viewHolder.itemView.context
-        val backgroundColor = ContextCompat.getColor(context, MESSAGE_DELETED_BACKGROUND)
-        val radius = dpToPx(DEFAULT_CORNER_RADIUS.toInt()).toFloat()
+        val backgroundColor = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_DELETED_BACKGROUND)
+        val radius = dpToPx(DEFAULT_CORNER_RADIUS_DP).toFloat()
+        val paint = Paint().apply {
+            color = backgroundColor
+            isAntiAlias = true
+            style = Paint.Style.FILL
+        }
         viewHolder.binding.deleteLabel.background = BackgroundDrawable(
-            color = backgroundColor,
+            paint = paint,
             topLeftCornerPx = radius,
             topRightCornerPx = radius,
             bottomRightCornerPx = if (data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius,
             bottomLeftCornerPx = radius
         )
     }
 
+    override fun decoratePlainTextMessage(
+        viewHolder: MessagePlainTextViewHolder,
+        data: MessageListItem.MessageItem
+    ) {
+        val radius = dpToPx(DEFAULT_CORNER_RADIUS_DP).toFloat()
+        val paint = if (data.isMine) {
+            Paint().apply {
+                style = Paint.Style.FILL
+                color = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_CURRENT_USER_BACKGROUND)
+                isAntiAlias = true
+            }
+        } else {
+            Paint().apply {
+                style = Paint.Style.STROKE
+                strokeWidth = dpToPx(DEFAULT_STROKE_WIDTH_DP).toFloat()
+                color = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_OTHER_STROKE_COLOR)
+                isAntiAlias = true
+            }
+        }
+        viewHolder.binding.messageText.background = BackgroundDrawable(
+            paint = paint,
+            topLeftCornerPx = radius,
+            topRightCornerPx = radius,
+            bottomRightCornerPx = if (data.isMine && data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius,
+            bottomLeftCornerPx = if (data.isMine.not() && data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius
+        )
```

Comment:
For me, the corners here are visibly thicker than the straight sections of the line:

![image](https://user-images.githubusercontent.com/12054216/100878637-be7ed580-34aa-11eb-904d-26845063f5eb.png)

Maybe the straight borders are hitting the layout bounds, and are being clipped off somehow?

Reply:
That's what makes me furious. Android doesn't offer a way to draw border "inside" the specified path. It draws it in both directions: half of the border width is painted inside the path and another half is painted outside. Very inconvenient.

Reply:
Look at previous comment